x-pack/plugins/rule_registry/common/parse_technical_fields.ts

```diff
@@ -8,9 +8,14 @@
 import { isLeft } from 'fp-ts/lib/Either';
 import { PathReporter } from 'io-ts/lib/PathReporter';
 import { technicalRuleFieldMap } from './assets/field_maps/technical_rule_field_map';
+import { experimentalRuleFieldMap } from './assets/field_maps/experimental_rule_field_map';
+
 import { runtimeTypeFromFieldMap } from './field_map';
 
-const technicalFieldRuntimeType = runtimeTypeFromFieldMap(technicalRuleFieldMap);
+const technicalFieldRuntimeType = runtimeTypeFromFieldMap({
+  ...technicalRuleFieldMap,
+  ...experimentalRuleFieldMap,
```

Comment:
Since these experimental fields are now solution specific, I don't think we want to include them in `technicalFieldRuntimeMap` - technical fields are used in all mappings

Reply:
Yep this doesn't seem ideal, I also didn't like but it was the easiest way to fix the typescript errors I was getting. In Observability there are quite a few places that call the `parseTechnicalFields` function. 

It looks like I will have to create an `experimentalFieldRuntimeMap` as well. Hopefully it won't require a lot of refactoring

Reply:
@marshallmain I pushed a new commit https://github.com/elastic/kibana/pull/120904/commits/adfd987265e874456f41227842ec39834d35aa6e. Does it look better now with ParsedTechnicalFields separate from ParsedExperimentalFields?

Now I have another question, that keeps bothering me. I kept the names for the experimental fields under `rule-data-utils` package but inside `technical_field_names`. I had tried to move them to a new experimental_field_names, but for some reason it failed. Do you recommend that I try again to move the definitions for the experimental field names in a separate file, right?

Reply:
Did it fail in CI or building locally? Often when adding or moving things around in packages I have to force the package to rebuild, a regular `yarn kbn bootstrap` doesn't seem to work all the time. `yarn kbn clean && yarn kbn bootstrap` sometimes fixes it for me.

Reply:
The separate `ParseExperiementalFields` looks good though, thanks for updating it!

Reply:
@marshallmain The last commit broke stuff in `create_lifecycle_executor`, and quite a few things are not rendering properly in Observability Alerts table. It is a parse error. I will look into this one, but it looks like this is a bigger refactoring that could break even more things. For example apm alert annotations use the parseTechnicalFields as well and I would need to adjust this as well and do a bit of testing to make sure I don't break more stuff. Would you mind if I merge this PR without last commit and work on refactoring and cleaning up experimental fields in a separate PR?

My concern is that most people will start vacation on Monday and I don't want to rush a change that would break stuff.

Reply:
Yeah we can leave the experimental fields in the `technicalFieldRuntimeType` for now and come back to it later on.

Reply:
@marshallmain I looked into it and it turns out that this commit fixes the Observability issues I faced https://github.com/elastic/kibana/pull/120904/commits/13f38255a4b3cfdcfda8d08d2a4fc0841e073b15.